test/integration/kubernetes_deploy_test.rb

```diff
@@ -616,6 +616,52 @@ def test_output_when_switching_labels_incorrectly
     ], in_order: true)
   end
 
+  def test_scale_down_deployment
+    dep_name = nil
+    pod_name = nil
+    pod_status = "Running"
+    source = 3
+    # Create a deployement with 3 pods
+    result = deploy_fixtures("hello-cloud", subset: ["configmap-data.yml", "web.yml.erb"]) do |fixtures|
+      web = fixtures["web.yml.erb"]["Deployment"].first
+      web["spec"]["replicas"] = source
+      # Disable grace period in deleting pods to speed up test
+      web["spec"]["template"]["spec"]["terminationGracePeriodSeconds"] = 0
+      dep_name = web["metadata"]["name"]
+      pod_name = web["spec"]["template"]["metadata"]["labels"]["name"]
+    end
+    assert_deploy_success(result)
+    hello_cloud = FixtureSetAssertions::HelloCloud.new(@namespace)
+    hello_cloud.assert_pod_status(pod_name, pod_status, 3)
+    # Scale down to 1 pod
```

Comment:
I don't think this step is really necessary. The problem seemed to be going from X replicas to zero. I don't have any reason to believe that 1 would be a special case.